src/pages/resultsView/fusion/ResultViewFusionMapperStore.ts

```diff
@@ -0,0 +1,190 @@
+import {
+    fetchEnsemblTranscriptsByEnsemblFilter,
+    fetchPfamDomainData,
+} from 'shared/lib/StoreUtils';
+import { PfamDomain } from 'genome-nexus-ts-api-client';
+import { Gene } from 'cbioportal-ts-api-client';
+
+export const FUS_GENE_COLORSCALE = [
+    '#097AFF',
+    '#3B9CFF',
+    '#28a9ff',
+    '#84B7DF',
+    '#92C8E8',
+    '#68ACEA',
+    '#4eb7d9',
+];
+export const REF_GENE_COLORSCALE = [
+    '#88C542',
+    '#99c556',
+    '#a0cf31',
+    '#7ba237',
+    '#47cb4a',
+    '#008828',
+    '#005623',
+];
+
+export class ResultViewFusionMapperStore {
+    constructor(
+        public gene: Gene,
+        public studyIdToStudy: MobxPromise<{ [studyId: string]: CancerStudy }>,
+        public molecularProfileIdToMolecularProfile: MobxPromise<{
+            [molecularProfileId: string]: MolecularProfile;
+        }>,
+        public samples: MobxPromise<SampleIdentifier[]>,
+        public fusions: StructuralVariantExt[]
+    ) {
+        makeObservable(this);
+        labelMobxPromises(this);
+    }
+
+    static addColor(transcripts: EnsemblTranscriptExt[], geneSymbol: string) {
+        const fusionTranscripts: EnsemblTranscriptExt[] = [];
+        return transcripts.map(t => {
+            // Check if transcript is reference gene, if yes then mark it
+            const isReferenceGene = t.hugoSymbols.indexOf(geneSymbol) >= 0;
+            // add color based on if transcript is reference gene or not
+            const fillColor = isReferenceGene
+                ? REF_GENE_COLORSCALE[0]
+                : FUS_GENE_COLORSCALE[
+                      fusionTranscripts.length % FUS_GENE_COLORSCALE.length
+                  ];
+            const newTranscript = Object.assign({}, t, {
+                isReferenceGene,
+                fillColor,
+            });
+            if (!isReferenceGene) {
+                const countRef = fusionTranscripts.filter(ref => {
+                    return ref['geneId'] === newTranscript['geneId'];
+                });
+                if (countRef.length === 0) {
+                    fusionTranscripts.push(newTranscript);
+                }
+            }
+            return newTranscript;
+        });
+    }
+
+    readonly ensemblTranscripts = remoteData(
+        {
+            invoke: async () => {
+                // Collect all transcript id's from the fusion data
+                const transcriptIds = new Set<string>();
+                this.fusions.forEach((fusion: StructuralVariant) => {
+                    transcriptIds.add(fusion.site1EnsemblTranscriptId);
+                    transcriptIds.add(fusion.site2EnsemblTranscriptId);
+                });
+                // Fetch ensembl transcript data with transcript ids as request params
+                return await fetchEnsemblTranscriptsByEnsemblFilter({
+                    transcriptIds: Array.from(transcriptIds),
+                }).then(result =>
+                    ResultViewFusionMapperStore.addColor(
```

Comment:
avoid using static functions because we don't do it very much and it isn't common in js.  just declare the functino above the class and call it